accelerator/http.c

```diff
@@ -149,11 +149,57 @@ static inline int process_get_tips_request(ta_http_t *const http,
   return set_response_content(ret, out);
 }
 
+static inline int process_send_transfer_request(ta_http_t *const http,
+                                                char const *const payload,
+                                                char **const out) {
+  status_t ret = SC_OK;
+  ret = api_send_transfer(&http->core->tangle, &http->core->service, payload,
+                          out);
+  return set_response_content(ret, out);
+}
+
+static inline int process_recv_mam_msg_request(ta_http_t *const http,
+                                               char const *const url,
+                                               char **const out) {
+  status_t ret = SC_OK;
+  char *bundle = NULL;
+  ret = ta_get_url_parameter(url, 1, &bundle);
+  if (ret == SC_OK) {
+    ret = api_receive_mam_message(&http->core->service, bundle, out);
+  }
+  return set_response_content(ret, out);
+}
+
+static inline int process_mam_send_msg_request(ta_http_t *const http,
+                                               char const *const payload,
+                                               char **const out) {
+  status_t ret = SC_OK;
+  ret = api_mam_send_message(&http->core->tangle, &http->core->service, payload,
+                             out);
+  return set_response_content(ret, out);
+}
+
+static inline int process_invalid_path_request(char **const out) {
+  cJSON *json_obj = cJSON_CreateObject();
+  cJSON_AddStringToObject(json_obj, "message", "Invalid path");
+  *out = cJSON_PrintUnformatted(json_obj);
+  return MHD_HTTP_BAD_REQUEST;
+}
+
+static inline int process_options_request(char **const out) {
+  cJSON *json_obj = cJSON_CreateObject();
+  cJSON_AddStringToObject(json_obj, "message", "OPTIONS request");
+  *out = cJSON_PrintUnformatted(json_obj);
+  return MHD_HTTP_OK;
+}
+
 static int ta_http_process_request(ta_http_t *const http, char const *const url,
-                                   char const *const payload,
-                                   char **const out) {
-  // TODO: TA api callback functions,
-  //      includes error response, handling option request
+                                   char const *const payload, char **const out,
+                                   int is_options) {
```

Comment:
I dislike the name `is_options`, which sounds weird. Thinks of something "is options". More than one "option"?